docs/react-testing-library/example-intro.md

```diff
@@ -11,27 +11,29 @@ See the following sections for a detailed breakdown of the test
 ```jsx
 // __tests__/fetch.test.js
 import React from 'react'
+import { rest } from 'msw'
+import { setupServer } from 'msw/node'
 import { render, fireEvent, waitFor, screen } from '@testing-library/react'
 import '@testing-library/jest-dom/extend-expect'
-import axiosMock from 'axios'
 import Fetch from '../fetch'
 
-jest.mock('axios')
+const server = setupServer(
+  rest.get('/greeting', (req, res, ctx) => {
+    return res(ctx.json({ greeting: 'hello there' }))
+  })
+)
+
+beforeAll(() => server.listen())
+afterAll(() => server.close())
```

Comment:
Should we add an `afterEach(() => server.resetHandlers())` here?

Reply:
As a safe option—yes, but unless you declare runtime handlers via `server.use()` there’s nothing to reset. I’d not suggest calling `resetHandlers` by default. Developers that need it will find it :)

Reply:
That's a fair point. Now that you mention it, I think the `server.use` call should be *inside* the test rather than outside it. That's what people should be doing most of the time.

With that in mind, resetHandlers should be included.

And I think it should be used by default 😉

Reply:
May I please double check on this?

I see your point of using resetHandlers and server.use inside a test suite, but I think I'm confused of what runtime handlers to add via server.use, as the example tests exactly one API communication to "GET /greeting". 

```js
const server = setupServer(
  // centralized place for request handlers,
  // generally encouraged.
  rest.get('/greetings', resolver)
)

beforeAll(...)
afterAll(...)

test('loads and displays greeting', () => {
  // I need `afterEach(() => server.resetHandlers())`
  // only if I use `server.use()` in any of my tests.
  server.use(/* however, what should I add here? */)
})
```

Do you see what I mean?

Reply:
This is a fair point. There are two things to consider:

1. Handle all of our application's typical requests so we don't have to think about them on a per-test basis
2. Add a handler for a specific test (to test an error state)

With that in mind, I think we should adjust this example so we have two tests. One for the success case, and one for the error case. That way the first one can rely on the default handlers and one that can add a runtime handler for the error.

Because I see using `afterEach(() => server.resetHandlers())` as an important default whenever someone uses `server.use` and we should probably demonstrate that so people don't add a bunch of complicated single-test-specific handlers to their `setupServer` call.

Reply:
_That_ is perfect.

Reply:
I've added the error handling test scenario to this example.

1. Rewrote `useState` to `useReducer` in `fetch.js` to reflect the error and data relation.
1. Added a new test suite, used `server.use()` with permanent override there.
1. Added `afterEach` that calls `server.resetHanders()`.

Please, could you double check if that `fetch.js` looks okay to you? Thanks.